src/test/java/com/redhat/devtools/lsp4ij/mock/MockTextDocumentService.java

```diff
@@ -384,7 +384,27 @@ public void setMockSelectionRanges(List<SelectionRange> mockSelectionRanges) {
 
     @Override
     public CompletableFuture<List<SelectionRange>> selectionRange(SelectionRangeParams params) {
-        return CompletableFuture.completedFuture(mockSelectionRanges);
+        // Find the mock selection ranges that apply to the specified position. This allows us to have a single mock
+        // response that covers multiple positions that might be queried during a given test.
+        List<SelectionRange> applicableMockSelectionRanges = mockSelectionRanges
+                .stream()
+                .filter(selectionRange -> {

```

Comment:
And this ensures that only the single selection range for the current location is returned from the list of mock selection ranges.